Z dimension parameter sweep visualization and results

gwaygenomics opened the pull request:
Similar to #119 - here are several visualizations and a write up about these results. Probably best to track in [`z_dimensions_hyperparameter_sweep_results.md`](https://github.com/gwaygenomics/tybalt/blob/86dac3dfd5d51ce807b354ef862c3cf81355b7c7/z_dimensions_hyperparameter_sweep_results.md).

jaclyn-taroni commented:
@gwaygenomics Can you make the palettes for your figures a bit more colorblind-friendly (see: https://github.com/clauswilke/colorblindr)? I probably should have brought this up in #119, too. I think the colors in the loss figures here are particularly important for interpretation.

jaclyn-taroni commented on scripts/viz/param_sweep_latent_space_viz.R:
```diff
@@ -0,0 +1,321 @@
+adage_param_z_png <- file.path(adage_fig, "z_parameter_adage.png")
+adage_param_z_pdf <- file.path(adage_fig, "z_parameter_adage.pdf")
+
+p <- ggplot(adage_select_df,
```
Can you reorder the levels in `sparsity` for this plot so that the legend is in increasing order? It took me a moment to square the plot with the text around it in `z_dimensions_hyperparameter_sweep_results.md` with the current ordering

gwaygenomics replied:
Yeah, good point. I will update the ordering in the next commit!

jaclyn-taroni commented on scripts/viz/param_sweep_latent_space_viz.R:
```diff
@@ -0,0 +1,321 @@
+# Several figures describing the results of a parameter sweep over latent space
+# dimensionality for both Tybalt and ADAGE models
+
+library(dplyr)
```
Since you use `dplyr::` quite a bit here, I'm assuming that this is just for the magrittr pipe? If so, can do:

```R
`%>%` <- dplyr::`%>%`
```

jaclyn-taroni commented on scripts/viz/param_sweep_latent_space_viz.R:
```diff
@@ -0,0 +1,321 @@
+p <- ggplot(tybalt_one_model, aes(x = as.numeric(paste(train_epoch)),
+                             y = val_loss)) +
+  geom_line(aes(color = kappa), size = 0.2, alpha = 0.7) +
+  facet_wrap(~num_components) +
```
Spacing around `~`

jaclyn-taroni commented on scripts/viz/param_sweep_latent_space_viz.R:
```diff
@@ -0,0 +1,321 @@
+                       `0.002` = "Learning: 0.002",
+                       `0.0025` = "Learning: 0.0025")
+
+base_theme <-  theme(axis.text = element_text(size = rel(0.5)),
```
In my opinion, I think this would be better placed before you read in files. Almost like it's a function. I found myself looking around for `base_theme` once I got to later plots.

gwaygenomics replied:
Yeah, I agree. Updated in next commit!

jaclyn-taroni commented on scripts/viz/param_sweep_latent_space_viz.R:
```diff
@@ -0,0 +1,321 @@
+# latent space dimensionality.
+
+# We observed that increasing the number of components resulted in changes to
+# optimal hyperparameters. As the dimensionaly increased, the learning rate
```
`dimensionaly` -> `dimensionality`

gwaygenomics commented:
Thanks @jaclyn-taroni - I've updated the color palette of this pull request. I will save updating the color palette of #119 for a later date (in discussing with @cgreene it actually may make sense for that to live in the CZI project repo)

I also addressed all other comments. thanks again!
